contributed/interpolate_trilinear_fwd.py

```diff
@@ -0,0 +1,344 @@
+"""
+
+import math
+import numpy as np
+import torch
+from torch import nn
+import torch_xla.core.xla_model as xm
+
+from neuronxcc import nki
+import neuronxcc.nki.language as nl
+import neuronxcc.nki.typing as nt
+from neuronxcc.nki import benchmark
+
+# For debugging
+# import os
+# os.environ["NEURON_FRAMEWORK_DEBUG"] = "1"
+# os.environ["NEURON_CC_FLAGS"]= " --disable-dge "
+
+
+@nki.jit
+def interpolate_trilinear_2x_fwd(src_arr: nt.tensor, chunk_size: int = 4) -> nt.tensor:
+    """
+    This function implements a NKI kernel for trilinear 2x upscaling (interpolation) on a 5D 
+    input tensor array for the `align_corners=False` case and using advanced indexing only.
+    Source array N & C dimensions are mapped to the SBUF partition dimension. D, H & W dimensions are mapped to the 
+    SBUF free dimension. The kernel chunks the source array along the D dimension in chunks of `chunk_size` faces. 
+    From a physical memory perspective, the kernel therefore operates on source array tiles of 
+    size (pmax, chunk_size * h_src * w_src).
+
+    Args:
+        src_arr (nt.tensor): Input HBM tensor of shape (N, C, D_src, H_src, W_src).
+        chunk_size (int): Size of the chunk along the D free dimension.
+    """
+    P_TILE_SIZE = nl.tile_size.pmax # 128
+
+    n_src, c_src, d_src, h_src, w_src = src_arr.shape
+    n_dst, c_dst, d_dst, h_dst, w_dst = n_src, c_src, d_src * 2, h_src * 2, w_src * 2
+
+    assert n_src == n_dst, "Input batch and output batch sizes must be identical"
+    assert c_src == c_dst, "Input channel and output channel sizes must be identical"
+    assert d_dst == 2 * d_src, "Output depth must be twice the input depth"
+    assert h_dst == 2 * h_src, "Output height must be twice the input height"
+    assert w_dst == 2 * w_src, "Output width must be twice the input width"
+    n, c = n_src, c_src
+    
+    weight_1d = 1 / 4 # specific to scaling_factor=2.0 & align_corners=False
+
+    src_arr = src_arr.reshape((n * c, d_src, h_src, w_src))
+    dst_arr = nl.ndarray(
+        [n_dst * c_dst, d_dst, h_dst, w_dst], 
+        dtype=src_arr.dtype,
+        buffer=nl.shared_hbm
+    )
+
+    wdw_size = chunk_size
+    step_size = wdw_size - 1
+
+    for d in nl.static_range(math.ceil((d_src - wdw_size)/step_size) + 1):
+        d_start_hbm_src = max(0, d * step_size) 
+        d_end_hbm_src = min(wdw_size + d * step_size, d_src)
+        
+        d_tile_size_src = d_end_hbm_src - d_start_hbm_src
+        d_tile_size_dst = 2 * d_tile_size_src
+        
+        d_start_tile_dst = 1 if d_start_hbm_src else 0
+        d_end_tile_dst = d_tile_size_dst
+        
+        d_start_hbm_dst = 2 * d_start_hbm_src + 1 if d_start_hbm_src else 0
+        d_end_hbm_dst = 2 * d_end_hbm_src
+        for p in nl.affine_range(math.ceil(n * c / P_TILE_SIZE)):
+            out_tile = nl.ndarray([P_TILE_SIZE, d_tile_size_dst, h_dst, w_dst], dtype=src_arr.dtype, buffer=nl.sbuf)
```

Comment:
refactor P_TILE_SIZE to free_dim or use a list of 2d tiles. block dimension is being deprecated